rest/data/pod.go

```diff
@@ -28,42 +28,77 @@ func (c *DBPodConnector) CreatePod(p restModel.APICreatePod) (*restModel.APICrea
 		}
 	}
 
-	return &restModel.APICreatePodResponse{ID: podDB.ID}, nil
+	return &model.APICreatePodResponse{ID: podDB.ID}, nil
 }
 
 // CheckPodSecret checks for a pod with a matching ID and secret in the
-// database.
+// database. It returns an error if the secret does not match the one assigned
+// for the given pod ID.
 func (c *DBPodConnector) CheckPodSecret(id, secret string) error {
 	p, err := pod.FindOneByID(id)
 	if err != nil {
-		return err
+		return gimlet.ErrorResponse{
+			StatusCode: http.StatusInternalServerError,
+			Message:    errors.Wrap(err, "finding pod by ID").Error(),
+		}
 	}
 	if p == nil {
-		return errors.New("pod does not exist")
+		return gimlet.ErrorResponse{
+			StatusCode: http.StatusNotFound,
+			Message:    "pod does not exist",
+		}
 	}
 	if secret != p.Secret {
 		return errors.New("incorrect pod secret")
```

Comment:
maybe we should return a 403?

Reply:
I fixed it. It's supposed to return 401.